webapp/channels/src/components/post_view/show_more/show_more.tsx

```diff
@@ -63,8 +65,34 @@ export default class ShowMore extends React.PureComponent<Props, State> {
         if (this.overflowRef) {
             window.cancelAnimationFrame(this.overflowRef);
         }
+        this.cleanupResizeObserver();
     }
 
+    setupResizeObserver = () => {
+        if (!this.textContainer.current || !window.ResizeObserver) {
+            // ResizeObserver is not supported in this browser or the container is not available yet
+            return;
+        }
+
+        this.cleanupResizeObserver();
```

Comment:
Why is this needed? In theory, `setupResizeObserver` only gets called on mount, so whatever the cleanup do, is already clean, right?

Reply:
It doesn't hurt to have it there, but I'd rather to keep it for several reasons, the main one for defensive programming and in case the component is modified later to call setupResizeObserver in response to prop changes or other events, the cleanup here will for sure make sure that we are not creating multiple observers unnecesarily. Either way I am 3/5 to keep it.